lib/stages/visRawReader.cpp

```diff
@@ -113,12 +114,30 @@ visRawReader::visRawReader(Config& config, const std::string& unique_name,
         _num_stack = _t.at("structure").at("num_stack").get<uint32_t>();
     }
 
+    // Match frequencies to IDs in the Telescope...
+    auto& tel = Telescope::instance();
+    std::map<double, uint32_t> inv_freq_map;
+
+    // ... first construct a map of central frequencies to IDs known by the
+    // telescope object
+    for (uint32_t id = 0; id < tel.num_freq(); id++) {
+        inv_freq_map[tel.to_freq(id)] = id;
+    }
+
+    // ... then use this to match the central frequencies given in the file
     for (auto f : freqs) {
-        // TODO: add freq IDs to raw file format instead of restoring them here
-        // TODO: CHIME specific.
-        uint32_t freq_id = 1024.0 / 400.0 * (800.0 - f.centre);
-        DEBUG("restored freq_id for f_centre={:.2f} : {:d}", f.centre, freq_id);
-        _freqs.push_back({freq_id, f});
+
+        auto it = inv_freq_map.find(f.centre);
+
+        if (it == inv_freq_map.end()) {
+            FATAL_ERROR("Could not match a frequency ID to channel in file at {} MHz. "
```

Comment:
It looks like this is getting triggered in one of the unit tests. `Could not match a frequency ID to channel in file at 798.828 MHz. Check you are specifying the correct telescope.`  Numerical precision issue?

Reply:
No. It was an embarrassing bug in the ICETelescope class. It was calculating the nyquist zones backwards!